src/value/mod.rs

```diff
@@ -491,13 +485,22 @@ pub trait Struct {
     ) -> Box<dyn Iterator<Item = &'a Self::Element> + 'a>;
 }
 
-/// Represents the constructor of Lob (i.e. `clob` and `blob`).
-pub trait LobBuilder {
+pub trait Builder {
     type Element: Element + ?Sized;
+    type Sequence: Sequence<Element = Self::Element> + ?Sized;
+
+    /// Build a `null` from IonType using Builder
+    fn new_null(e_type: IonType) -> Self;
+
+    /// Build a `clob` using Builder
+    fn new_clob(bytes: &'static [u8]) -> Self;
+
+    /// Build a `blob` using Builder
+    fn new_blob(bytes: &'static [u8]) -> Self;
 
-    /// build `clob` from Lob
-    fn into_clob(self) -> Self::Element;
+    /// Build a `list` from Sequence using Builder
+    fn new_list(seq: Self::Sequence) -> Self;
 
-    /// build `blob` from Lob
-    fn into_blob(self) -> Self::Element;
+    /// Build a `sexp` from Sequence using Builder
+    fn new_sexp(seq: Self::Sequence) -> Self;
```

Comment:
The more I think about this, the more I wonder if this should just all return `Self::Element`, I think this made sense as `Self` when this was `LobBuilder`, but I am not so sure now, what do you think?  It also means we don't need an `Into<Element>` trait implementation (which adds a cycle anyhow).

If we keep with this factoring, we will need some kind of `into_element()` for this to work generically to go from `Builder` to `Element`.

If we do go straight to `Self::Element` you can implement it for `...Element` and avoid returning an `...Value` as the builder intermediate.

Reply:
Yes. I think we should go with `Self::Element` as the return type here rather than returning intermediate `....Value`.